packages/gitbook/src/components/primitives/Input.tsx

```diff
@@ -0,0 +1,313 @@
+ * Input component with core functionality (submitting, clearing, validating) and shared styles.
+ */
+export const Input = React.forwardRef<HTMLInputElement | HTMLTextAreaElement, InputProps>(
+    (props, passedRef) => {
+        const {
+            multiline,
+            value: passedValue,
+            sizing = 'medium',
+            tag = 'div',
+            leading,
+            trailing,
+            className,
+            clearButton,
+            submitButton,
+            submitMessage,
+            label,
+            'aria-label': ariaLabel,
+            'aria-busy': ariaBusy,
+            placeholder,
+            keyboardShortcut,
+            disabled,
+            onSubmit,
+            onChange,
+            onKeyDown,
+            containerRef,
+            maxLength,
+            minLength,
+            resize = false,
+            ...rest
+        } = props;
+
+        const [value, setValue] = React.useState(passedValue ?? '');
+        const [submitted, setSubmitted] = React.useState(false);
+        const inputRef = React.useRef<HTMLInputElement | HTMLTextAreaElement>(null);
+        const ref = passedRef ?? inputRef;
+
+        const language = useLanguage();
+        const isControlled = 'value' in props;
+        const hasValue = value.toString().trim();
+        const hasValidValue =
+            hasValue &&
+            (maxLength ? value.toString().length <= maxLength : true) &&
+            (minLength ? value.toString().length >= minLength : true);
+
+        const sizes = {
+            medium: {
+                container: `${multiline ? 'p-2' : 'px-4 py-2'} gap-2 circular-corners:rounded-3xl rounded-corners:rounded-xl`,
+                input: '-m-2 p-2',
+                gap: 'gap-2',
+            },
+            large: {
+                container: `${multiline ? 'p-3' : 'px-6 py-3 '} gap-3 circular-corners:rounded-3xl rounded-corners:rounded-xl`,
+                input: '-m-3 p-3',
+                gap: 'gap-3',
+            },
+        };
+
+        const handleChange = (event: React.ChangeEvent<HybridInputElement>) => {
+            const newValue = event.target.value;
+            if (!isControlled) {
+                setValue(newValue);
+            }
+            onChange?.(event);
+
+            // Reset submitted state when user edits the value to allow re-submission
+            if (submitted) {
+                setSubmitted(false);
+            }
+
+            if (multiline && resize && 'current' in ref && ref.current) {
+                ref.current.style.height = 'auto';
+                ref.current.style.height = `${ref.current.scrollHeight}px`;
+            }
+        };
+
+        const handleClear = () => {
+            if (!('current' in ref) || !ref.current) return;
+
+            const syntheticEvent = {
+                target: { value: '' },
+                currentTarget: ref.current,
+            } as React.ChangeEvent<HybridInputElement>;
+
+            handleChange(syntheticEvent);
```

Comment:
Bad idea. My suggestion for that is to create a `onInputChange` callback that provides only the value. And then you call safely call it here. But faking the change event is not good.